### PR TITLE
Parse nationality

### PR DESCRIPTION
**Pull request by st-devabit**

### CODE REVIEW

File: src/parse/__tests__/td2.js [hunk 1]
```diff
@@ -34,4 +34,56 @@ describe('parse TD2', () => {
...
+    expect(result.fields.lastName).toStrictEqual('KOZLOVSKA8');
+  });
+
+  it('space in document number', () => {
+    const MRZ = [
+      'I<UTOERIKSSON<<ANNA<MARIA<<<<<<<<<<<',
+      '592988362<<<<7408122F12041590804<<<6'
+    ];
+    const result = parse(MRZ);
+    expect(result.fields.documentNumber).toStrictEqual('592988362080');
```
Comment (lukesolo) [inline src/parse/__tests__/td2.js:86]: Document number can be separated this way?

File: src/parsers/parseDocumentNumberCheckDigit.js [hunk 2]
```diff
@@ -12,7 +12,11 @@ module.exports = function parseDocumentNumberCheckDigit(
     const tail = optional.substring(0, firstFiller - 1);
     source = `${source}<${tail}`;
     checkDigit = optional.charAt(firstFiller - 1);
-    check(source, checkDigit);
+    try {
+      check(source, checkDigit);
+    } catch (e) {
+      check(source.replace(/</g, ''), checkDigit);
```
Comment (lukesolo) [inline src/parsers/parseDocumentNumberCheckDigit.js]: What is it the case? Check digit can be missing?

File: src/parsers/parseSex.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ module.exports = function parseSex(source) {
     case 'F':
       return 'female';
     case '<':
-      return 'nonspecified';
```
Comment (lukesolo) [inline src/parsers/parseSex.js]: We can use this const in AC

File: src/parsers/parseDocumentNumberCheckDigit.js (hunk 2 above)
Comment (lukesolo) [reply src/parsers/parseDocumentNumberCheckDigit.js]: Anyway there is no point to use catch to clean the source. It's better to do it beforehand.

**Review by lukesolo — CHANGES_REQUESTED** [review]

File: src/parsers/parseSex.js (hunk 3 above)
Comment (st-devabit) [reply src/parsers/parseSex.js]: @geruz  requested to return an empty string.

File: src/parsers/parseDocumentNumberCheckDigit.js (hunk 2 above)
Comment (st-devabit) [reply src/parsers/parseDocumentNumberCheckDigit.js]: document number can contain filler and in some cases check digit should be calculated with this character, in another - without.

File: src/parse/__tests__/td2.js (hunk 1 above)
Comment (st-devabit) [reply src/parse/__tests__/td2.js:86]: yes, according to documentNumberTemplate in td2
`{
    line: 1,
    start: 0,
    end: 9,
    related: [
      {
        line: 1,
        start: 9,
        end: 10
      },
      {
        line: 1,
        start: 28,
        end: 35
      }
    ]
  }`

**Review by lukesolo — APPROVED** [review]

**Pull request merged**